main.go

```diff
@@ -1,93 +1,107 @@
-    "fmt"
-    "log"
-    "os"
+	"bufio"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
 )
 
-func seedColumnIds() []string{
-    columnIds := make([]string, 0)
-    f, err := os.Open("columnIds.txt")
+func seedColumnIds() []string {
+	columnIds := make([]string, 0)
+	f, err := os.Open("columnIds.txt")
 
-    if err != nil {
-        log.Fatal(err)
-    }
+	if err != nil {
+		log.Fatal(err)
+	}
 
-    defer f.Close()
+	defer f.Close()
 
-    scanner := bufio.NewScanner(f)
+	scanner := bufio.NewScanner(f)
 
-    for scanner.Scan() {
+	for scanner.Scan() {
 
-        //fmt.Println(scanner.Text())
-        columnIds = append(columnIds, scanner.Text())
-    }
+		//fmt.Println(scanner.Text())
+		columnIds = append(columnIds, scanner.Text())
+	}
 
-    if err := scanner.Err(); err != nil {
-        log.Fatal(err)
-    }
-    return columnIds
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	return columnIds
 }
 
 func main() {
-    var wg sync.WaitGroup
-    work := make(chan string, 10000)
-
-    var columnIdsToDelete []string = seedColumnIds()
-    fmt.Println(len(columnIdsToDelete))
-
-    // write columnIds to the work channel, blocking until a worker goroutine
-    // is able to start work
-    for _, columnId := range columnIdsToDelete {
-        fmt.Println(columnId)
-        work <- columnId
-    }
-
-    // startup pool of 10 go routines and read columnIds from work channel 
-    for {
-        for i := 0; i<=10; i++ {
-            wg.Add(1)
-            go func(w chan string) {
-                defer wg.Done()
-                columnId := <-w
-                url := fmt.Sprintf("https://api.honeycomb.io/1/columns/vector_multitenant_prod_filtered/%s", columnId)
-                sendRequest(url)
-                time.Sleep(1*time.Second)
-                fmt.Println(url)
-            }(work)
-        }
-    }
-
-    wg.Wait()
-
-    fmt.Println("Done")
+	var wg sync.WaitGroup
+	work := make(chan string)
+
+	var columnIdsToDelete []string = seedColumnIds()
+	fmt.Println(len(columnIdsToDelete))
+
+	// write columnIds to the work channel, blocking until a worker goroutine
+	// is able to start work
+	go func(ids []string) {
+		for _, columnId := range ids {
+			fmt.Println(columnId)
+			work <- columnId
+		}
+		// close channel to tell workers data is complete
+		close(work)
+	}(columnIdsToDelete)
+
+	// Start up pool of goroutines and read columnIds from work channel.
+	// Honeycomb doesn't explicitly list a rate limit for column operations,
+	// but Events are limited to 2000/s for free tier, 7000/s for pro/enterprise.
+	// https://intercom.help/honeycomb/en/articles/2121867-is-there-a-limit-to-how-fast-i-can-send-in-data
+	// To compromise between speed for deleting up to thousands of columns and polite usage of the API,
+	// we limit to 100 column deletions per second.
+	for i := 0; i < 100; i++ {
+		wg.Add(1)
+		go func(w chan string) {
+			defer wg.Done()
+			for {
+				select {
+				case columnId, ok := <-w:
```

Comment:
Reading from a closed channel won't cause an error, but it will return `false` for `ok`, so we know to break out of our infinite loop.